### PR TITLE
allow underscore for network names, not endpoint name

### PR DESCRIPTION
**Pull request by timcharper**
JIRA Issue: MARATHON-7848

### CODE REVIEW

File: src/test/scala/mesosphere/ValidationTestLike.scala [hunk 1]
```diff
@@ -17,6 +19,20 @@ import org.scalatest.matchers.{ BePropertyMatchResult, BePropertyMatcher, MatchR
 trait ValidationTestLike extends Validation {
   this: Assertions =>
 
+  /**
+    * Validator which takes an object, serializes it to JSON, and then parses it back, allowing up to test validations
```
Comment (kensipe) [inline src/test/scala/mesosphere/ValidationTestLike.scala]: "allowing up to test" doesn't make any sense

File: src/main/scala/mesosphere/marathon/state/AppDefinition.scala [hunk 2]
```diff
@@ -573,7 +573,6 @@ object AppDefinition extends GeneralPurposeCombinators {
     // constraints are only validated in RAML layer
     appDef.unreachableStrategy is valid
     appDef.networks is valid(NetworkValidation.modelNetworksValidator)
-    appDef.networks is every(NetworkValidation.modelNetworkValidator)
```
Comment (timcharper) [inline src/main/scala/mesosphere/marathon/state/AppDefinition.scala]: all of the network-level validation is handled in RAML now

**Comment by timcharper** [comment]
@kensipe it is called "round trip" because we serialize it to JSON, and then parse it back, and check for violations.

I am open for naming suggestions. I used roundTrip because it was there before.

**Comment by kensipe** [comment]
I read the javadocs and understand what it does... It is a crappy name and I did give several naming suggestions.

**Pull request merged**